### ISSUE TITLE
literature review

### ISSUE THREAD

**Issue by jerry-shijieli** [azhe825/CSC510#4]
## since 2015:
- Improving Automated Email Tagging with Implicit Feedback (Amrit)
- A New Algorithm to Categorize E-mail Messages to Folders with Social Networks Analysis (Di)
- Alecsa: Attentive Learning for Email Categorization using Structural Aspects (Zhe)
- Adaptive Ant Colony Decision Forest in Automatic Categorization of Emails (SHijie)
## highly cited
- Generating summary keywords for emails using topics (Amrit)
- Summarizing spoken and written conversations (Di)
- BuzzTrack: topic detection and tracking in email(Zhe)
- Methods for mining and summarizing text conversations(Shijie)
## goals
- data source 
- methods (best solutions)
- future works especially for paper since 2015
- motivation application

**Comment by jerry-shijieli** [comment]
Methods for mining and summarizing text conversations(Shijie)
This is a book, not a paper. Let me read the following paper instead:
- Intelligent Email: Aiding Users with AI (Shijie)

**Comment by azhe825** [comment]
## Alecsa: Attentive Learning for Email Categorization using Structural Aspects
- Decision Fusion (different models to predict different type of folders, then combine the prediction together)
- attentive learning (3 type of folders: content-aware, time-aware, participant-aware)
- has a lot of related work description
- performance metric: average accuracy
- Data source: Use the following in bash:

```
wget http://people.cs.umass.edu/~ronb/datasets/enron_flat.tar.gz
```

**Comment by azhe825** [comment]
## BuzzTrack: topic detection and tracking in email
- Clustering (unsupervised learning)
- Labeling after clustering by using most common words in the subjects
- Introduction is useful, describes why we need to regroup inbox, why we don't want to manually group it...
- TDT evaluation: New Topic detection and topic tracking.
- Performance metric: precision, recall, FP, FN, TP. FP and FN forms a DET curve.
- Data: not Enron, data collected by their own, at ETH Zurich

**Comment by dichen001** [comment]
## Automatic Categorization of Email into Folders: Benchmark Experiments on Enron and SRI Corpora

### Useful notes for our proposal
- Email folders do not necessarily correspond to simple semantic topics
- They correspond to unfinished todo tasks, project groups, certain recipients, or loose agglomerations of topics.
- Users tend to folder some messages by sender, some by event, some by date and in many cases the logic behind a certain foldering choice can be difficult to discern.
- Email content and foldering habits differ drastically from one email user to another.
- An email foldering system should be adaptive to the working style of individual users.

### Data source
- Use the well-known [Enron Email Dataset](http://www.cs.cmu.edu/~enron/)
- A smaller but foldered corpus [SRI](http://www.ai.sri.com/project/CALO). 
- However, the link for downloading data can not be found anymore. Because the author changed his domain of his homepege. We may write email for the data.

See [this link](http://management.haifa.ac.il/images/info_people/ron_bekkerman_files/document%20classification%20on%20enron%20email%20dataset.pdf) for the data provided by the author. The author's homepage is now at [here](http://management.haifa.ac.il/index.php/en/2013-01-16-13-45-34/info?id=513:rbekkerman/).

### Method and Results
- Comparing email classification results of four classifiers (Maximum Entropy, Naive Bayes, SVM and Winnow),
- SVM performs best for most of the cases.
- Winnow classifier is computationally attractive and easy to implement, 
- Winnow classifier in many cases achieving statistically similar performance as SVM
- Evaluation: dividing a dataset into time-based data chunks and incrementally testing classi¯cation performance on each chunk while training on all the previous chunks.

**Comment by dichen001** [comment]
Changed the paper from **Adaptive regularization of weight vectors** to  **Summarizing spoken and written conversations**.

**Comment by jerry-shijieli** [comment]
## Intelligent Email: Aiding Users with AI

### Data Source:

Enron data set

### Summary:

The authors take a user-oriented approach to apply AI to the email classifications. The classifications focus on three tasks: summary keyword generation, email reply prediction and attachment prediction.

### Methods:
- For the summary keyword generation, unsupervised approach based on latent concept models(LSA and LDA) of a user’s mailbox is used, because they requires no annotated training data and generates keywords that describe each message in the context of other related messages in user’s mailbox.
- For the reply prediction, the authors introduce user-specific relational features constructed from user profiles. 
- For attachment prediction, relational features generated from user profiles and additional task-specific features in emails are involved to predict the needs of attachments.

### Evaluation:

Recall, Precision, F1 factor. Also cross-user evaluation to test performance on new user.

**Comment by dichen001** [comment]
## A New Algorithm to Categorize E-mail Messages to Folders with Social Networks Analysis
- They improve the accuracy with which emails are assigned to folders.
- Ant Colony Optimization (ACO) algorithms and Social Networks Analysis (SNA) are used.
- Also use the Enron Email dataset.
- Final results show their algorithm almost outperform all the algorithms in accuracy.

**Comment by jerry-shijieli** [comment]
# Adaptive Ant Colony Decision Forest in Automatic Categorization of Emails

### Data source:

Enron Email Dataset

### Methods:

The authors transform the email dataset into a decision table using the Ant Colony Decision Forest(ACDF algorithm which is an Ensemble classifier based on the Ant Colony Decision Trees. Each splitting rule of the the decision trees is performed independently for different subset of attributes. And Bagging and Boosting strategy is used for training process. Ant Colony Optimization is a type of AI which is a form of emergent collection intelligence of groups of simple individuals.

### Evaluation:
- The evaluations of decision trees in the training stage will be calculated using Equation(8) which is related to the tree size, and classification accuracy of that tree from training dataset. 
- The Experiment results compares AdaBoost, Bagging, Random Forest, and adaptive ACDF with aACDF giving the highest accuracy.

### Problem And Future Work:

There is no explicit discussion about the data processing or feature extraction. And the authors state in the futrure stages, the text mining elements in analysis of email content will be incorporated with the pheromne trail of ACO algorithm to produce positive effects.

**Comment by dichen001** [comment]
## Summarizing Spoken and Written Conversations
- Not THAT related to our topic.
- Take extractice approach to summarization.
- For all ML experiments, utilize logistic regression classifiers. 
- Use conversational features in a ML sentence classfication framework.
- Evaluation metrics: pyramid precision and weighted f-score.
- Data:
  - For the email summarization, it is on the Enron corpus.
  - For the conversation part, it uses AMI corpus.
- Results: 
  - The two most useful features are sentence length and CWS. 
  - The lexical and length features are the most effective feature subclasses.

**Comment by amritbhanu** [comment]
## Improving Automated Email Tagging with Implicit Feedback

(A really great paper for our application)

### Data Source
- They created the datasets on their own by collecting through various web sources. No link provided.

### Motivation
- Unprioritized mails
- Manual organization of emails

### Current Application
- Gmail organize mails using combination of some rules.
- An **Email Predictor 2** plugin is already there which is a Microsoft Outlook add - on.
- Various features are already provided.
  - There will be initial mails and tags associated with those are taken as training data.
  - After prediction, users can give an Explicit feedback and correct the tags.
  - If user made a mistake in the tag, rollback option is given.
  - If classifier is not confident then positive feedback will help in correcting those predictions

### Assumptions:
- User spend time for longer on emails and changes the tags, so if he corrects the tag then, the remaining tags must be correct.

### Methods
- No Implicit Feedback (NoIF) : After prediction, users can just edit/remove/add the tags. Negative prediction and positive prediction will stay with the same level of confidence for the tags.
- Simple Implicit Feedback (SIF) : After prediction, users can just edit/remove/add the tags. Negative prediction will gain the confidence and positive prediction will gain the confidence. That it will consider all the tags as correctly.
  - Implicit Feedback without SIF (IFwoSIF) : Counts number of implicit feedback events (like copying, replying, forwarding, saving an attachment, moving to a folder). When count reaches specific threshold, then training examples are immediately created.
  - Implicit Feedback with SIF (IFwSIF) : If the user changes a tag, then implicit feedback examples are immediately created. Else, it continues to count up events until the number of events reach a threshold.

### Results
- IFwSIF performs better. 

### Future Paper which can be read
- Adaptive regularization of weight vectors
- Confidence-Weighted Linear Classification

**Comment by amritbhanu** [comment]
## Generating summary keywords for emails using topics

## Dataset
- Enron

## Methods
- Unsupervised learning
  - LSA doc query
  - LSA word association
  - LDA doc query
  - LDA word association

## Application
- Recipient Prediction
- Automated Foldering

## Results
- LDA doc out performs all others as well as TF-IDF
- Evaluating parameter was Precision and accuracy

## Future Work
- Working with other latent models.

It can be helpful when we create summary of let say 9-10 keywords to distinguish among each emails. The most common words can be assigned a single "label" by us and say all the mails which include these keywords will correspond to that "label".

**Comment by jerry-shijieli** [comment]
Just checked the Enron email data set source:
http://www.cs.cmu.edu/~enron/
But the size is not 423MB, it is actually 1.82GB in *.tgz or 2.72GB after decompressing.

**Comment by amritbhanu** [comment]
Jerry can you open it and check how are the folders organised? With recipients name - label -emails

**Comment by jerry-shijieli** [comment]
Yes, here are some screenshots of the folders and original email file.

![screen shot 2016-01-30 at 4 43 12 pm](https://cloud.githubusercontent.com/assets/7918460/12698670/f711edbc-c770-11e5-98ec-f96bcf165e92.png)

![screen shot 2016-01-30 at 4 43 48 pm](https://cloud.githubusercontent.com/assets/7918460/12698671/004b43e2-c771-11e5-9381-77d72ce7f215.png)

![screen shot 2016-01-30 at 4 44 23 pm](https://cloud.githubusercontent.com/assets/7918460/12698672/06312be6-c771-11e5-9a5c-87e0a64ed0e0.png)

![screen shot 2016-01-30 at 4 44 52 pm](https://cloud.githubusercontent.com/assets/7918460/12698673/0aef1314-c771-11e5-9513-c00051f069b4.png)

![screen shot 2016-01-30 at 4 47 50 pm](https://cloud.githubusercontent.com/assets/7918460/12698684/451bcc76-c771-11e5-82bc-89e71b362a6e.png)
